### PR TITLE
The last collectible is shown for a moment while opening a new collectible #18750

### LINKED ISSUE #18750: The last collectible is shown for a moment while opening a new collectible
Original report: https://github.com/status-im/status-mobile/pull/18532#issuecomment-1909279658

> In the case of collectible, we are seeing last collectible because we are requesting data from status-go.
> In my testing, I only saw status-go calls differ in traits, etc, other than that we already have required data in db that we can use instead of waiting from status-go.

Note: Please checkout https://github.com/status-im/status-mobile/pull/18532#issuecomment-1910502741 https://github.com/status-im/status-mobile/pull/18532#issuecomment-1911736872 before fixing

[299521223-20b4fde1-1028-49d5-b17a-9741cf42c0d0.webm](https://github.com/status-im/status-mobile/assets/17097240/7e5a6301-ac36-477f-bc29-586cf74a4514)

### PR DESCRIPTION
**Pull request by mmilad75**
fixes #18750 

This PR clears the data saved on `:last-collectible-details` when the collectible screen is closed.

https://github.com/status-im/status-mobile/assets/55688834/1d100877-6de1-4cec-b532-6126500921cb

### CODE REVIEW

**Comment by ulisesmac** [comment]
Hey @mmilad75 

I can see in this PR, we show a blank screen for a while.

We can reduce the exposure to that screen (and actually probably solve it) by triggering the navigation event at the right moment.

Currently, when we press a collectible, we:

1. Perform an rpc request and navigate to the new screen
2. Store the collectible details

The blank screen can be reduced if we change the order of the events to:
1. Perform an rpc request
2. Store the collectible details and navigate


Since this PR is too related I think it's a place to address it.

**Comment by mmilad75** [comment]
> Hey @mmilad75
> 
> I can see in this PR, we show a blank screen for a while.
> 
> We can reduce the exposure to that screen (and actually probably solve it) by triggering the navigation event at the right moment.
> 
> Currently, when we press a collectible, we:
> 
> 1. Perform an rpc request and navigate to the new screen
> 2. Store the collectible details
> 
> The blank screen can be reduced if we change the order of the events to:
> 
> 1. Perform an rpc request
> 2. Store the collectible details and navigate
> 
> Since this PR is too related I think it's a place to address it.

@ulisesmac Done. Please have another look

**Review by J-Son89 — APPROVED** [review]

File: src/status_im/contexts/wallet/events/collectibles.cljs [hunk 1]
```diff
@@ -122,7 +122,8 @@
          {:keys [collectibles]} response
          collectible            (first collectibles)]
      (if collectible
-       {:fx [[:dispatch [:wallet/store-last-collectible-details collectible]]]}
+       {:fx [[:dispatch [:wallet/store-last-collectible-details collectible]]
+             [:dispatch [:navigate-to :wallet-collectible]]]}
        (log/error "failed to get collectible details"
```
Comment (ulisesmac) [inline src/status_im/contexts/wallet/events/collectibles.cljs]: Hey @mmilad75 

thanks for applying the suggestion, but I think we could delay the navigation even more, inside the event that is storing:

image

wdyt?

**Review by ulisesmac — APPROVED** [review]

File: src/status_im/contexts/wallet/events/collectibles.cljs (hunk 1 above)
Comment (mmilad75) [reply src/status_im/contexts/wallet/events/collectibles.cljs]: Sure 👍 Let me apply it

**Review by smohamedjavid — APPROVED** [review]

File: src/status_im/contexts/wallet/events/collectibles.cljs (hunk 1 above)
Comment (mmilad75) [reply src/status_im/contexts/wallet/events/collectibles.cljs]: Done @ulisesmac

**Comment by VolodLytvynenko** [comment]
hi @mmilad75 thank you for PR. Take a look please at found issue. For me, it's reproducible only on IOS

Unfortunately, there are no exact steps to reproduce it. It happens randomly to me. I hope the logs might help understand what's going on.

## ISSUE 1: Collectible is opened twice on iOS

### Steps:
1. Restore the user with available collectibles.
2. Open the wallet.
3. Open a particular ETH account where the collectible is located.
4. Open the collectible.

### Actual result:
The collectible is opened twice.

https://github.com/status-im/status-mobile/assets/52490791/dec937be-e60a-4aa8-b1c9-de4e67431653

### Expected result: 
The collectible is opened once.

### Device:
- Real Device iPhone 11 Pro max, IOS 17

### Logs:
[logs (4).zip](https://github.com/status-im/status-mobile/files/14296916/logs.4.zip)

**Comment by ulisesmac** [comment]
> hi @mmilad75 thank you for PR. Take a look please at found issue. For me, it's reproducible only on IOS
> 
> Unfortunately, there are no exact steps to reproduce it. It happens randomly to me. I hope the logs might help understand what's going on.
> 
> ## ISSUE 1: Collectible is opened twice on iOS
> 
> ### Steps:
> 1. Restore the user with available collectibles.
> 2. Open the wallet.
> 3. Open a particular ETH account where the collectible is located.
> 4. Open the collectible.
> 
> ### Actual result:
> The collectible is opened twice.
> 
> https://github.com/status-im/status-mobile/assets/52490791/dec937be-e60a-4aa8-b1c9-de4e67431653
> 
> ### Expected result: 
> The collectible is opened once.
> 
> ### Device:
> - Real Device iPhone 11 Pro max, IOS 17
> 
> ### Logs:
> [logs (4).zip](https://github.com/status-im/status-mobile/files/14296916/logs.4.zip)
> 

@mmilad75 

It seems we are navigating twice 👀 make sure we only have one call.

File: src/status_im/contexts/wallet/collectible/view.cljs [hunk 2]
```diff
@@ -116,6 +116,10 @@
             {collection-image :image-url
              collection-name  :name}    collection-data
             {collectible-name :name}    collectible-data]
+        (rn/use-effect
+         (fn []
+           #(rf/dispatch [:wallet/clear-last-collectible-details]))
+         [])
```
Comment (vkjr) [inline src/status_im/contexts/wallet/collectible/view.cljs:122]: I think this empty vector can be omitted

**Review by vkjr — APPROVED** [review]

File: src/status_im/contexts/wallet/collectible/view.cljs (hunk 2 above)
Comment (mmilad75) [reply src/status_im/contexts/wallet/collectible/view.cljs]: Nice, I didn't know about this. Done

**Comment by mmilad75** [comment]
@VolodLytvynenko Hey man. Please have another look. The issue is fixed

File: src/status_im/contexts/wallet/collectible/view.cljs (hunk 2 above)
Comment (mmilad75) [reply src/status_im/contexts/wallet/collectible/view.cljs:122]: I want to run the code only in componentWillUnmount

**Comment by VolodLytvynenko** [comment]
Hi @mmilad75. Thank you for PR. No issues from my side. PR is ready to be merged

**Pull request merged**